### PR TITLE
SymInt'ify sparse_compressed_tensor 

### PR DESCRIPTION
**Pull request by pearu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #107903
* #115586

cc @ezyang @albanD @zou3519 @gqchen @nikitaved @soulitzer @Lezcano @Varal7

### CODE REVIEW

**Comment by pearu** [comment]
@albanD @soulitzer could you review this PR?

File: tools/autograd/derivatives.yaml [hunk 1]
```diff
@@ -1906,7 +1906,10 @@
 - name: values(Tensor(a) self) -> Tensor(a)
   dispatch:
     Default:
-      self: at::_sparse_coo_tensor_unsafe_symint(self.indices(), grad, self.sym_sizes())._coalesced_(true)
+      self: "grad.defined() ? (self.layout() == c10::kSparse) ? at::_sparse_coo_tensor_unsafe_symint(self.indices(), grad, self.sym_sizes(), self.options(), /*is_coalesced=*/true) :
```
[the remark this answers is not in the document]
Comment (pearu) [reply tools/autograd/derivatives.yaml]: Done.

File: tools/autograd/derivatives.yaml [hunk 2]
```diff
@@ -1888,7 +1888,7 @@
   indices: non_differentiable
   values: grad.sparse_mask(result)._values()
 
-- name: sparse_compressed_tensor.comp_plain_value_size(Tensor compressed_indices, Tensor plain_indices, Tensor values, int[] size, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=False) -> Tensor
+- name: sparse_compressed_tensor.comp_plain_value_size(Tensor compressed_indices, Tensor plain_indices, Tensor values, SymInt[] size, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=False) -> Tensor
```
[the remark this answers is not in the document]
Comment (pearu) [reply tools/autograd/derivatives.yaml:1891]: I followed the analogy of values backward support for sparse COO tensors, but indeed, it is not necessary here.
I'll refactor values backward support into a separate PR in this stack.

[the remark this answers is not in the document]
Comment (pearu) [reply tools/autograd/derivatives.yaml:1891]: Done: https://github.com/pytorch/pytorch/pull/115586

**Review by cpuhrsch — APPROVED** [review]

**Pull request closed without merging**